lib/src/collections.dart

```diff
@@ -13,8 +13,56 @@ import 'package:dart_kollection/src/collection/set_mutable.dart';
 
 /**
  * Returns a new read-only list of given elements.
+ *
+ * Elements aren't allowed to be `null`. If your list requires `null` values use [listFrom]
  */
-KList<T> listOf<T>([Iterable<T> elements = const []]) {
+KList<T> listOf<T>(
+    [T arg0,
+    T arg1,
+    T arg2,
+    T arg3,
+    T arg4,
+    T arg5,
+    T arg6,
+    T arg7,
+    T arg8,
+    T arg9]) {
+  List<T> args;
+  if (arg9 != null) {
+    args = [arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9];
+  } else if (arg8 != null) {
+    args = [arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8];
+  } else if (arg7 != null) {
+    args = [arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7];
+  } else if (arg6 != null) {
+    args = [arg0, arg1, arg2, arg3, arg4, arg5, arg6];
+  } else if (arg5 != null) {
+    args = [arg0, arg1, arg2, arg3, arg4, arg5];
+  } else if (arg4 != null) {
+    args = [arg0, arg1, arg2, arg3, arg4];
+  } else if (arg3 != null) {
+    args = [arg0, arg1, arg2, arg3];
+  } else if (arg2 != null) {
+    args = [arg0, arg1, arg2];
+  } else if (arg1 != null) {
+    args = [arg0, arg1];
+  } else if (arg0 != null) {
+    args = [arg0];
+  } else {
+    return emptyList();
```

Comment:
const constructor?

Reply:
Glad to help. 👿 
That's too bad. But in any case, since the list is immutable then always returning the same class instance even if not `const`,  should be fine right?

Reply:
I can't return `KList<dynamic>` for `KList<String>`

```
KList _empty = EmptyList();
KList<T> emptyList<T>() => _empty;

// error: type 'EmptyList<dynamic>' is not a subtype of type 'KList<String>'
emptyList<String>()
```